tests/lib/ast-utils.js

```diff
@@ -0,0 +1,507 @@
+                assert.equal(sourceCode, "// my line comment");
+            }
+
+            var config = { rules: {} },
+                spy = sandbox.spy(handler);
+
+            eslint.reset();
+            eslint.on("LineComment", spy);
+
+            eslint.verify(code, config, filename, true);
+            assert(spy.calledOnce, "Handler should be called.");
+        });
+
+        it("should fire LineComment and LineComment:exit events", function() {
+
+            function handler(node) {
+                var sourceCode = eslint.getSource(node);
+                assert.equal(node.value, " my line comment");
+                assert.equal(sourceCode, "// my line comment");
+            }
+
+            var config = { rules: {} },
+                spy = sandbox.spy(handler);
+
+            eslint.reset();
+            eslint.on("LineComment", spy);
+            eslint.on("LineComment:exit", spy);
+
+            eslint.verify(code, config, filename, true);
+            assert(spy.calledTwice, "Handler should be called.");
+        });
+
+        it("should fire BlockComment event", function() {
+
+            function handler(node) {
+                var sourceCode = eslint.getSource(node);
+                assert.equal(node.value, " my block comment ");
+                assert.equal(sourceCode, "/* my block comment */");
+            }
+
+            var config = { rules: {} },
+                spy = sandbox.spy(handler);
+
+            eslint.reset();
+            eslint.on("BlockComment", spy);
+
+            eslint.verify(code, config, filename, true);
+            assert(spy.calledOnce, "Handler should be called.");
+        });
+
+        it("should fire LineComment event for top-level comment", function() {
+
+            function handler(node) {
+                var sourceCode = eslint.getSource(node);
+                assert.equal(node.value, " fixme");
+                assert.equal(sourceCode, "// fixme");
+            }
+
+            var config = { rules: {} },
+                spy = sandbox.spy(handler);
+
+            eslint.reset();
+            eslint.on("LineComment", spy);
+
+            eslint.verify("// fixme", config, filename, true);
+            assert(spy.calledOnce, "Handler should be called.");
+        });
+
+        it("should fire BlockComment:exit event", function() {
+
+            function handler(node) {
+                var sourceCode = eslint.getSource(node);
+                assert.equal(node.value, " my block comment ");
+                assert.equal(sourceCode, "/* my block comment */");
+            }
+
+            var config = { rules: {} },
+                spy = sandbox.spy(handler);
+
+            eslint.reset();
+            eslint.on("BlockComment", spy);
+            eslint.on("BlockComment:exit", spy);
+
+            eslint.verify(code, config, filename, true);
+            assert(spy.calledTwice, "Handler should be called.");
+        });
+
+    });
+
+});
```

Comment:
1. Are all of these tests really related to ast-utils? Seems like some of them are testing that eslint fires events for comment nodes. 
2. A lot of ast-utils functions are not tested at all (e.g. isSpaced)